packages/arb-token-bridge-ui/src/state/app/utils.ts

```diff
@@ -47,7 +52,43 @@ export const getDepositStatus = (tx: Transaction | MergedTransaction) => {
     return DepositStatus.L1_PENDING
   }
 
-  // l1 succeeded...
+  // for teleport txn
+  if (
+    isTeleport({
+      sourceChainId: tx.parentChainId, // we make sourceChain=parentChain assumption coz it's a deposit txn
+      destinationChainId: tx.childChainId
+    }) &&
+    tx.l1ToL2MsgData &&
+    tx.l2ToL3MsgData
+  ) {
+    const { l2ToL3MsgData, l1ToL2MsgData } = tx
+
+    // if we find `l2ForwarderRetryableTxID` then this tx will need to be redeemed
+    if (l2ToL3MsgData?.l2ForwarderRetryableTxID) return DepositStatus.L2_FAILURE
+
+    // else, normally check the status of l3Retryable and l2Retryable
+    switch (l2ToL3MsgData.status) {
+      case L1ToL2MessageStatus.CREATION_FAILED:
+        return DepositStatus.CREATION_FAILED
+      case L1ToL2MessageStatus.EXPIRED:
+        return DepositStatus.EXPIRED
+      case L1ToL2MessageStatus.FUNDS_DEPOSITED_ON_L2:
+        return DepositStatus.L2_FAILURE
+      case L1ToL2MessageStatus.REDEEMED:
+        return DepositStatus.L2_SUCCESS
+    }
+    switch (l1ToL2MsgData.status) {
+      case L1ToL2MessageStatus.CREATION_FAILED:
+        return DepositStatus.CREATION_FAILED
+      case L1ToL2MessageStatus.EXPIRED:
+        return DepositStatus.EXPIRED
+      case L1ToL2MessageStatus.FUNDS_DEPOSITED_ON_L2:
+        return DepositStatus.L2_FAILURE
+      case L1ToL2MessageStatus.REDEEMED:
+        return DepositStatus.L2_PENDING // tx is still pending if l1ToL2MsgData is redeemed
+    }
```

Comment:
I couldn't tell that the final deposit status is different between the two closures after staring at it for 5 minutes, so I think this is very bad for debugging

```suggestion
    const depositStatus = getDepositStatusFromL1ToL2MessageStatus(
      l2ToL3MsgData.status
    )
    if (typeof depositStatus !== 'undefined') {
      return depositStatus
    }
    switch (l1ToL2MsgData.status) {
      case L1ToL2MessageStatus.REDEEMED:
        return DepositStatus.L2_PENDING // tx is still pending if l1ToL2MsgData is redeemed
      default:
        return getDepositStatusFromL1ToL2MessageStatus(l1ToL2MsgData.status)
    }
```

```typescript

// somewhere in this file
function getDepositStatusFromL1ToL2MessageStatus(
  status: L1ToL2MessageStatus
): DepositStatus | undefined {
  switch (status) {
    case L1ToL2MessageStatus.CREATION_FAILED:
      return DepositStatus.CREATION_FAILED
    case L1ToL2MessageStatus.EXPIRED:
      return DepositStatus.EXPIRED
    case L1ToL2MessageStatus.FUNDS_DEPOSITED_ON_L2:
      return DepositStatus.L2_FAILURE
    case L1ToL2MessageStatus.REDEEMED:
      return DepositStatus.L2_SUCCESS
  }
}
```